packages/context/src/objects/objects.ts

```diff
@@ -62,48 +132,172 @@ export type ObjectInstance = {
+};
+
+export type Content = {
+    type: 'content';
+    html: string;
+    visibility?: VisibilityConfiguration | string;
+};
+
+export type Column = {
+    width: number;
+    entries?: FormEntry[];
+};
+
+export type Columns = {
+    type: 'columns';
+    columns: Column[];
+    visibility?: VisibilityConfiguration | string;
+};
+
+export type Section = {
+    label: string;
+    entries?: FormEntry[];
+};
+
+export type Sections = {
+    type: 'sections';
+    sections: Section[];
+    visibility?: VisibilityConfiguration | string;
+};
+
+export type FormEntry = InputParameterReference | Columns | Sections | Content;
+
+export type Form = {
+    entries?: FormEntry[];
+};
+
 /**
  * Represents an object action inputProperty object.
  */
 export type ActionInput = {
-    label: string;
-    type: string;
-    key: string;
-    initialValue?: unknown;
+    id?: string;
+    label?: string;
+    type?:
+        | 'button'
+        | 'Section'
+        | 'Columns'
+        | 'Content'
+        | 'Select'
+        | 'TextField'
+        | 'DateTime'
+        | 'RepeatableField'
+        | 'MultiSelect'
+        | 'Decimal'
+        | 'RichText'
+        | 'Date'
+        | 'Integer'
+        | 'Image'
+        | 'Object'
+        | 'Time'
+        | 'User';
+    key?: string;
+    initialValue?: string | number | SelectOption[] | SelectOption;
+    defaultToCurrentDate?: boolean;
+    defaultToCurrentTime?: boolean;
+    html?: string;
+    labelPosition?: string;
     placeholder?: string;
     description?: string;
     tooltip?: string;
     prefix?: string;
     suffix?: string;
-    showCharCount?: boolean;
-    readOnly?: boolean;
-    isMultiLineText?: boolean;
     data?: {
         /**
          * An array of values required for select options.
          */
         values?: SelectOption[];
     };
+    inputMask?: string;
+    inputMaskPlaceholderChar?: string;
+    tableView?: boolean;
+    mode?: 'default' | 'existingOnly';
+    displayOption?: 'dropdown' | 'dialogBox';
+    rows?: number;
+    showCharCount?: boolean;
+    readOnly?: boolean;
+    isMultiLineText?: boolean;
+    verticalLayout?: boolean;
+    input?: boolean;
+    widget?: string;
+    conditional?: {
+        json?: string;
+        show?: boolean;
+        when?: string;
+        eq?: string | number;
+    };
+    property?: {
+        id: string;
+    };
     validate?: {
         required?: boolean;
+        ciriteria?: object;
         operator?: 'any' | 'all';
-        regexes?: RegexValidation[];
+        regexes?: { regex: string; errorMessage?: string }[];
+        minLength?: number;
+        maxLength?: number;
+        minDate?: string;
+        maxDate?: string;
+        minTime?: string;
+        maxTime?: string;
+        min?: number;
+        max?: number;
+        customMessage?: string;
     };
     /**
      * An array of sub-components to be rendered inside sections.
      */
-    components?: ActionInput[];
-    [key: string]: unknown;
```

Comment:
Removing this is technically a breaking change.  If we want to do it, I'm fine with it, but it should be marked as such in the commits so the versions are updated appropriately.

Reply:
Got it. I'll rebase to update the commit.